main/lsp/LSPPreprocessor.cc

```diff
@@ -119,121 +98,179 @@ bool LSPPreprocessor::ensureInitialized(LSPMethod method, const LSPMessage &msg)
     return false;
 }
 
-void LSPPreprocessor::preprocessAndEnqueue(QueueState &state, unique_ptr<LSPMessage> msg, absl::Mutex &stateMtx) {
-    ENFORCE(owner == this_thread::get_id());
-    if (msg->isResponse()) {
-        return;
-    }
-
-    const LSPMethod method = msg->method();
-    if (!ensureInitialized(method, *msg)) {
-        // msg is invalid. Error sent to client.
-        return;
+unique_ptr<LSPTask> LSPPreprocessor::getTaskForMessage(LSPMessage &msg) {
+    if (msg.isResponse()) {
+        // We currently send no requests to the client, so we don't expect any response messages.
+        return make_unique<SorbetErrorTask>(
+            *config, make_unique<SorbetErrorParams>((int)LSPErrorCodes::MethodNotFound, "Unexpected response message"));
     }
-
-    auto &logger = config->logger;
-    bool shouldEnqueue = false;
-    bool shouldMerge = false;
-    switch (method) {
```

Comment:
The interlacing in the diff is weird. This red logic is from the old `preprocessAndEnqueue`. I've moved all of this logic into the corresponding LSPTask classes on their `preprocess` methods.

For example, the LSPTask for CancelRequest calls `LSPPreprocessor::cancel` in its `preprocess` method.